### DISCUSSION TITLE
Reconsider equivalence between type.broadcastable and type.shape

### DISCUSSION THREAD

**Discussion by ricardoV94** [aesara-devs/aesara#1170] [Ideas]
Right now `TensorType("float64", shape=(None, 5, 1)).broadcastable == (False, False, True)`

However, due to enabling dynamic broadcasting, this is not correct as that type can broadcast along the first dimension when that has a size of 1 at runtime. In many rewrites we check for conditions like `var1.dtype.broadcastable == var2.dtype.broadcastable`, but we shouldn't assume they are the same when they come from `None` type shape dimensions. 

One option would be to change to the following: `TensorType("float64", shape=(None, 5, 1)).broadcastable == (np.nan, False, True)`, or any other type that respects `(self == self) is False` as `np.nan` does.

Related to #1089 and #1122

**Comment by brandonwillard** [comment]
Don't forget that **we're trying to phase out use of `TensorType.broadcastable` altogether**.  The word "broadcastable" with its obvious interpretation is not correct (and never was) in all cases, and that was a big part of why we wanted to replace it with a quantity that accurately represents the relevant information: i.e. `TensorType.shape`.  

The only reason `TensorType.broadcastable` is still relevant is due to us not having all the old code refactored yet.

**Reply by ricardoV94** [reply]
I know that, but many old rewrites relied on the older interpretation of broadcastable which is not the same as the new one.
